features/manage-position/Redeem.tsx

```diff
@@ -24,7 +29,8 @@ const Redeem = () => {
   const [error, setError] = useState<Error | null>(null);
 
   const tokensToRedeemFloat = isNaN(parseFloat(tokensToRedeem)) ? 0 : parseFloat(tokensToRedeem);
-  const maxRedeem = Math.min((syntheticBalance || 0), (borrowedTokens || 0));
+  const tokensAboveMin = borrowedTokens && empState.minSponsorTokens ? borrowedTokens - weiToNum(empState.minSponsorTokens) : 0;
+  const maxRedeem = Math.min((syntheticBalance || 0), (borrowedTokens || 0), (tokensAboveMin ||0));
```

Comment:
Does this prevent them from redeeming _all_ of their tokens?

Reply:
Yes, I believe so. Should we make an exception for that? Basically the rule should only apply if they are not unwinding their entire position?

Reply:
Currently, say `minSponsorTokens` is 1000, and they hold 1200 tokens, then:

- `syntheticBalance` = 1200
- `borrowedTokens` = 1200
- `tokensAboveMin` = 200

They would only be able to redeem 200.

Reply:
I think we should make an exception for that. They should be able to completely unwind their position IMO.

Reply:
So in this case we would want to basically say, you can redeem between 0 and 200 tokens or all 1200 tokens. This makes the logic pretty annoying, though :/

Reply:
lol yes this is very annoying.